gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -476,33 +475,26 @@ impl GatewayLightning for ClnRpcService {
 
     async fn route_htlcs(
         &self,
-        request: tonic::Request<tonic::Streaming<InterceptHtlcResponse>>,
+        _: tonic::Request<EmptyRequest>,
     ) -> Result<tonic::Response<Self::RouteHtlcsStream>, Status> {
-        let mut stream = request.into_inner();
-
         // First create new channel that we will use to send responses back to gatewayd
         let (gatewayd_sender, gatewayd_receiver) =
             mpsc::channel::<Result<InterceptHtlcRequest, Status>>(100);
 
         let mut sender = self.interceptor.sender.lock().await;
         *sender = Some(gatewayd_sender.clone());
 
-        // Spawn new thread that listens for events from the input stream
-        let interceptors = self.interceptor.clone();
-        tokio::spawn(async move {
-            while let Some(res) = stream.next().await {
-                if let Ok(complete_request) = res {
-                    let _ = Self::complete_htlc(complete_request, interceptors.clone())
-                        .await
-                        .map_err(|e| {
-                            error!("CLN extension failed to complete HTLC: {:?}", e);
-                        });
-                }
-            }
-        });
-
         Ok(tonic::Response::new(ReceiverStream::new(gatewayd_receiver)))
     }
+
+    async fn complete_htlc(
```

Comment:
The code you deleted was already calling `Self::complete_htlc`, so ... how can you add this method here without conflicting with an existing one ... . I don't see deleting an existing one ...

I'm so confused.

Reply:
Here we are implementing `GatewayLightning for ClnRpcService`. The existing method is directly on `ClnRpcService`.

Rust traits are fun :smile:

Reply:
Yeah perhaps I should just move the existing `complete_htlc` functionality into here to eliminate the confusion

Reply:
Moved this logic so that there's only one `complete_htlc`